lib/newrelic-roda/instrument.rb

```diff
@@ -23,20 +23,37 @@ def match_all(args)
           end
           add_method_tracer :match_all
 
+          def filter_params(param_hash)         
+            filtered_params = ENV['NEW_RELIC_RODA_FILTERED_PARAMS']&.split(',')
+            param_hash.with_indifferent_access.except(*filtered_params)
+          end
+
           def build_params
             rv = nil
-            body_pos = body.pos
+            body_pos = body.respond_to?(:pos) ? body.pos : 0
```

Comment:
We need to clean up the below code to make sure:

1. no if-else path gets filter called on it
2. that we don't call params and body.read unless absolutely necessary - both are very expensive methods if called on a big body
3. that stuff that could make us leak pin blocks, tracks etc is heavily tested - at least in unit tests here

Let's scale this down - we keep this PR - we continue working on it - but lets send a quick & dirty version that doesn't send params upstream